hathor/event/event_manager.py

```diff
@@ -193,8 +199,35 @@ def _create_event(
         return BaseEvent.from_event_arguments(
             event_id=0 if self._last_event is None else self._last_event.id + 1,
             peer_id=self._peer_id,
-            timestamp=self._clock.seconds(),
+            timestamp=self._reactor.seconds(),
             event_type=event_type,
             event_args=event_args,
             group_id=group_id,
         )
+
+    def _handle_load_started(self):
+        self._event_storage.save_node_state(NodeState.LOAD)
+
+    def _handle_load_finished(self):
+        self._event_storage.save_node_state(NodeState.SYNC)
+
+    def _should_reload_events(self) -> bool:
+        return self._previous_node_state in [None, NodeState.LOAD]
+
+    def handle_load_phase_vertices(self, topological_iterator: Iterator[BaseTransaction]) -> None:
+        """
+        Either generates load phase events or not, depending on previous node state.
+        Does so asynchronously so events generated here are not processed before normal event handling.
+        """
+        assert self._is_running, 'Cannot handle load phase events, EventManager is not started.'
+
+        if not self._should_reload_events():
+            return
+
+        for vertex in topological_iterator:
```

Comment:
I'm still uncomfortable with iterating over the whole database, as the full node already has to do it to start itself. My discomfort happens because it is a very slow operation for large blockchains. Couldn't the iterator on HathorManager call this method while it iterates over the vertices?

Reply:
I'm actually OK with this being a separate iterator loop, as long as the loop only happens when there are events to be emitted and this happens after the indexes initialization. My rationale is that if these events do need to be emitted it will be O(N) anyway (for N events), and if the indexes are initialized, an optimal iterator with no warmup will be able to be used, so at the end of the day it does seem like it will follow `N * (A + B)` -> `N * A + N * B` for A being a indexes initialization step and B the event emitting step. What do you think? @msbrogli @glevco

Reply:
I agree it works for now. I think in the future when we decide to refactor the iterator out of `IndexesManager`, we should take this into account and optimize it into a single loop. But considering your big O comments, I think it's fine for now.

> as long as the loop only happens when there are events to be emitted and this happens after the indexes initialization

Yep, that is the case

Reply:
I partially agree with the complexity analysis. Even though it will be N * (A + B), the iterator must load the transactions twice, and loading might be slower than emitting the events.

Anyway, I agree that we do like this for now. If the full node loading time increases too much, we can change it later.

Reply:
Agreed. Thanks!